src/CQRS/LeanCode.CQRS.AspNetCore/ServiceCollectionCQRSExtensions.cs

```diff
@@ -85,4 +85,12 @@ public CQRSServicesBuilder AddCQRSObjects(TypesCatalog contractsCatalog, TypesCa
         );
         return this;
     }
+
+    public CQRSServicesBuilder WithLocalCommands(Action<ICQRSApplicationBuilder> configure)
+    {
+        Services.AddSingleton<Local.ILocalCommandExecutor>(
+            s => new Local.MiddlewareBasedLocalCommandExecutor(s, configure)
```

Comment:
Two things there:
1. I wonder if it's good that the same command may have different local and 'remote' pipeline
2. We can have multiple 'remote' pipelines, but only one local?

These are not top priority things, but at some point we'll need to address them.

Reply:
1. That's something I also wonder myself, but considering how different are these modes, then I am leaning in favor of splitting them (because some middlewares won't make sense in local calls, some won't make sense in remote calls).
2. Yeah, this will need a solution.

Reply:
Okay, after some thought, this is what I came up with:
1. Local and remote pipelines are really different. It is up to the user to make it work in both. These can be the same, but it should be a deliberate choice (as, e.g., HTTP cache does not make any sense in local calls).
2. Keyed services to the rescue. I'm thinking of somewhat simplified `IHttpClientFactory` for local executors. Also, I think the key should be the only differentiator and we should allow all objects to go through it _by default_ (otherwise it might get tricky to explain to users how it works).

Reply:
Both done.